project-addons/flask_middleware_connector/models/stock.py

```diff
@@ -32,11 +35,35 @@ class StockMove(models.Model):
 
     @api.multi
     def write(self, vals):
-        res = super(StockMove, self).write(vals)
-        if vals.get('state', False) and vals["state"] != "draft":
-            for move in self:
-                session = ConnectorSession(self.env.cr, self.env.uid,
+        """
+        Incluir la comprobacion de todas las lineas del albaran despues de esta funcion.
+        Si un albaran tiene alguna linea esperando disponibilidad, su estado sera parcialmente disponible.
+        Si un albaran tiene todas las lineas listas, ninguna esperando disponibilidad y ninguna o \
+        varias lineas canceladas, su estado tiene que ser listo para transeferir. 
+        SI un albaran tiene todas sus lineas canceladas su estado pasa a cancelado. 
+        Si un albaran tiene todas sus lineas esperando disponibilidad y ninguna\
+        o alguna linea cancelada, su estado sera esperando disponibilidad  
+        """
+        old_state = False
+        new_state = False
+        res = False
+        for move in self:
+            old_state = self[0].picking_id.state
+            res = super(StockMove, self).write(vals)
```

Comment:
¿Por qué metes el super dentro del for? Ponlo fuera como estaba antes, sino en todo caso sería: super(StockMove, move).write(vals) peor no es necesario ya que vals no varía

Reply:
Ese write cambia el estado del picking, si lo saco fuera del for tengo que sacar old_state también y si saco esto tengo que hacer un bucle solo para cada old_state. Duplicaría el mismo bucle, voy a ver si lo puedo hacer de otra forma.

Reply:
No, fuera sólo sacas el new_state, el otro lo obtienes de cada movimiento, ya que pueden ser de distintos albaranes y tener estados distintos, lo que puedes hacer es tener un "set" de albaranes visitados, y en cada iteracción añades los pickings no cancelados de la venta, el propio set evitarña tener duplicados. Luego, debajo del for, puedes recorrerte este set de albaranes para llamara  fire en cada uno.